source/extensions/filters/udp/dns_filter/dns_filter_resolver.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/network/dns.h"
+
+#include "extensions/filters/udp/dns_filter/dns_parser.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+
+enum class DnsFilterResolverStatus { Pending, Complete, TimedOut };
+
+/*
```

Comment:
You will need to have a circuit breaker / limit on the number of pending queries. It's fine to do in a follow up but please add a TODO.

Reply:
Makes sense